src/familiar.ts

```diff
@@ -439,84 +446,44 @@ export function setMarginalFamiliar(loc: Location, underwater: boolean): void {
     .map((fam): MarginalFamiliar => {
-      const isLep = fam.leprechaunMultiplier > 0;
-      let additionalValue = 0;
-      if (isLep) {
-        if (!lepOutfitMeatPercent || !lepOutfitFamWeight) {
-          print("Computing leprechaun outfits for the first time...", "blue");
-          useFamiliar(fam.familiar);
-          meatOutfit(false, noFreeFightsReq, false);
-          lepOutfitMeatPercent = sum(outfitSlots, (slot: Slot) =>
-            getModifier("Meat Drop", equippedItem(slot))
-          );
-          lepOutfitFamWeight = sum(outfitSlots, (slot: Slot) =>
-            getModifier("Familiar Weight", equippedItem(slot))
-          );
-
-          print(`lepOutfitMeatPercent: ${lepOutfitMeatPercent}%`, "blue");
-          print(`lepOutfitFamWeight: ${lepOutfitFamWeight}lbs`, "blue");
-        }
-        additionalValue =
-          (toInt(barf) *
-            (lepOutfitMeatPercent +
-              numericModifier(
-                fam.familiar,
-                "Meat Drop",
-                familiarWeight(fam.familiar) + buffFamWeight + lepOutfitFamWeight,
-                underwaterFamEquip(fam.familiar, underwater)
-              )) *
-            baseMeat) /
-          100;
-      } else {
-        if (!nonLepOutfitMeatPercent || !nonLepOutfitFamWeight) {
-          print("Computing non-leprechaun outfits for the first time...", "blue");
-          useFamiliar(fam.familiar);
-          meatOutfit(false, noFreeFightsReq, false);
-          nonLepOutfitMeatPercent = sum(outfitSlots, (slot: Slot) =>
-            getModifier("Meat Drop", equippedItem(slot))
-          );
-          nonLepOutfitFamWeight = sum(outfitSlots, (slot: Slot) =>
-            getModifier("Familiar Weight", equippedItem(slot))
-          );
-
-          print(`nonLepOutfitMeatPercent: ${nonLepOutfitMeatPercent}%`, "blue");
-          print(`nonLepOutfitFamWeight: ${nonLepOutfitFamWeight}lbs`, "blue");
-        }
-        additionalValue = (toInt(barf) * nonLepOutfitMeatPercent * baseMeat) / 100;
-      }
+      const { meat } =
+        cachedOutfits.get(fam.leprechaunMultiplier) ??
+        cacheOutfit(fam.leprechaunMultiplier, fam.familiar);
+      const additionalValue =
+        (toInt(barf) * (meat + getFamModifier(fam, "Meat Drop")) * locBaseMeat +
+          getFamModifier(fam, "Item Drop") * 0.15 * 3 * 200) /
+        100;
       return {
         ...fam,
         marginalValue: fam.expectedValue + additionalValue,
       };
     })
     .sort((left, right) => right.marginalValue - left.marginalValue);
 
-  const nominalOutfitValue = ((lepOutfitMeatPercent ?? 0) * baseMeat) / 100;
+  let nominalOutfitValue: number | null = null;
+  cachedOutfits.forEach((value) => {
+    if (!nominalOutfitValue || (value.meat * locBaseMeat) / 100 < nominalOutfitValue) {
+      nominalOutfitValue = (value.meat * locBaseMeat) / 100;
+    }
+  });
+
   print("Considering Marginal Familiars:", "blue");
   dropFamiliars.forEach((fam) => {
```

Comment:
It looks like this code currently does not account for the bonus equips in each outfit? I could be missing where that's happening

Reply:
Ah, in my head I was thinking that we were always going to be equipping the same bonus equips (katana, mafia ring etc) that it wouldn't matter (since I normalize it with the `nominalOutfitValue`).

I guess this could matter in cases where we don't actually use the same bonus equips across different outfit.